src/gametest/java/ca/teamdman/sfm/gametest/SFMCorrectnessGameTests.java

```diff
@@ -1356,8 +1367,34 @@ public static void falling_anvil_xp_shard_many(GameTestHelper helper) {
         var pos = helper.absoluteVec(new Vec3(1.5, 3.5, 1.5));
         ItemStack enchBook = EnchantedBookItem.createForEnchantment(new EnchantmentInstance(
                 Enchantments.SHARPNESS,
-                3
+                4
         ));
+        EnchantedBookItem.addEnchantment(enchBook, new EnchantmentInstance(Enchantments.BLOCK_EFFICIENCY, 2));
+
+        var cases = List.of(
+                Pair.of(SFMConfig.Common.LevelsToShards.JustOne, 10),
+                Pair.of(SFMConfig.Common.LevelsToShards.EachOne, 20),
+                Pair.of(SFMConfig.Common.LevelsToShards.SumLevels, 60),
+                Pair.of(SFMConfig.Common.LevelsToShards.SumLevelsScaledExponentially, 100)
+        );
+
+        var restore = SFMConfig.COMMON.levelsToShards.get();
+
+        // kill old item entities
+        helper.getLevel().getEntitiesOfClass(ItemEntity.class, new AABB(helper.absolutePos(new BlockPos(1, 4, 1))).inflate(3)).forEach(e -> e.discard());
+        falling_anvil_xp_shard_many_inner(helper, restore, pos, enchBook, cases.iterator());
+    }
+
+    private static void falling_anvil_xp_shard_many_inner(GameTestHelper helper, SFMConfig.Common.LevelsToShards restore, Vec3 pos, ItemStack enchBook, Iterator<Pair<SFMConfig.Common.LevelsToShards, Integer>> iter) {
```

Comment:
not quite sure what's going on here

Reply:
recursion because runAfterDelay doesnt block